### DISCUSSION TITLE
[Possible Bug] Workflow not executed for pushes to PR branch when base branch contains "+"

### DISCUSSION THREAD

**Discussion by poikilotherm** [community/community#25695] [Actions]
<p>Maybe it’s just me. And I couldn’t find anything about this via search. Forgive me if I did sth wrong.</p>
<p>I have a GH action that should run on pushes and pull requests to a base branch named “develop+ct”.</p>
<p>When filtering for this branch name in on.pull_request.branches, the workflow never gets executed.<br>
It works however using an “if” condition.</p>
<p>Please see <a href="https://github.com/gdcc/dataverse/pull/20/commits/96dfe37174b298c769bfbd815005b572bddffab0" class="inline-onebox" rel="noopener nofollow ugc">More container by poikilotherm · Pull Request #20 · gdcc/dataverse · GitHub</a> for an example.</p>
<p>Is this a bug, a feature or just me being not able to read docs? Thanks for helping!</p>

**Comment by shrink** [comment, marked as answer]
<p>The inputs for branch filters aren’t strings, they’re regular expressions (“filter patterns”) of which <code>+</code> is a special character.</p>
<blockquote>
<p>You can use special characters in path, branch, and tag filters.<br>
…<br>
<code>+</code> : Matches one or more of the preceding character.</p>
</blockquote>
<p>The pattern you’ve used would match <code>developct</code>, <code>developpct</code>, <code>developppct</code>,<code> developpppct</code> etc.</p>
<p>A complete outline of the filter patterns is listed at the very bottom of the documentation under “<a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet" rel="noopener nofollow ugc">Filter pattern cheat sheet</a>” – not obvious if you’re not looking for it!</p>
<p>The pattern you’re looking for to match <code>develop+ct</code> exactly is:</p>
<pre><code class="lang-auto">develop\+ct
</code></pre>
<p><a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet" class="onebox" target="_blank" rel="noopener nofollow ugc">https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet</a></p>

**Comment by poikilotherm** [comment]
<p>D’oh!</p>
<p>Hidden in plain sight. Thank you for bumping my nose into it!</p>
<p>Do you think it would be worse adding some notes about this? (like “take care to escape when branch names contain glob chars from the filter pattern matching”) I’d open a PR for that…</p>

**Comment by Simran-B** [comment]
<p>You can definitely propose such a change in the docs repo and see what the documentation team thinks:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/github/docs">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/github/docs" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/7819991/181092537-5f21cb4c-51b8-400d-8e97-b318a2906bfd.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/5/4/54476eaff403db2f17b8120dcf64e3db600f1bff_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/5/4/54476eaff403db2f17b8120dcf64e3db600f1bff_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/5/4/54476eaff403db2f17b8120dcf64e3db600f1bff.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/5/4/54476eaff403db2f17b8120dcf64e3db600f1bff_2_10x10.png"></div>

<h3><a href="https://github.com/github/docs" target="_blank" rel="noopener nofollow ugc">github/docs</a></h3>

  <p>The open-source repo for docs.github.com. Contribute to github/docs development by creating an account on GitHub.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by poikilotherm** [comment]
<aside class="onebox githubpullrequest" data-onebox-src="https://github.com/github/docs/pull/8107">
  <header class="source">

      <a href="https://github.com/github/docs/pull/8107" target="_blank" rel="noopener nofollow ugc">github.com/github/docs</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Pull Request">
    <svg width="60" height="60" class="github-icon" viewbox="0 0 12 16" aria-hidden="true"><path d="M11 11.28V5c-.03-.78-.34-1.47-.94-2.06C9.46 2.35 8.78 2.03 8 2H7V0L4 3l3 3V4h1c.27.02.48.11.69.31.21.2.3.42.31.69v6.28A1.993 1.993 0 0 0 10 15a1.993 1.993 0 0 0 1-3.72zm-1 2.92c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zM4 3c0-1.11-.89-2-2-2a1.993 1.993 0 0 0-1 3.72v6.56A1.993 1.993 0 0 0 2 15a1.993 1.993 0 0 0 1-3.72V4.72c.59-.34 1-.98 1-1.72zm-.8 10c0 .66-.55 1.2-1.2 1.2-.65 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2zM2 4.2C1.34 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/github/docs/pull/8107" target="_blank" rel="noopener nofollow ugc">Add clarification about glob chars in branch/tag matching</a>
    </h4>

    <div class="branches">
      <code>github:main</code> ← <code>poikilotherm:patch-2</code>
    </div>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2021-07-12" data-time="07:54:34" data-timezone="UTC">07:54AM - 12 Jul 21 UTC</span>
      </div>

      <div class="user">
        <a href="https://github.com/poikilotherm" target="_blank" rel="noopener nofollow ugc">
          <img alt="poikilotherm" src="https://user-images.githubusercontent.com/5468333/181092538-492cb9dd-4571-4f51-a927-20a17148ec89.jpeg" class="onebox-avatar-inline" width="20" height="20">
          poikilotherm
        </a>
      </div>

      <div class="lines" title="1 commits changed 1 files with 1 additions and 1 deletions">
        <a href="https://github.com/github/docs/pull/8107/files" target="_blank" rel="noopener nofollow ugc">
          <span class="added">+1</span>
          <span class="removed">-1</span>
        </a>
      </div>
    </div>
  </div>
</div>

  <div class="github-row">
    <p class="github-body-container">The docs weren't obvious pointing out that it's more than `*` and `**` that migh<span class="show-more-container"><a href="https://github.com/github/docs/pull/8107" target="_blank" rel="noopener nofollow ugc" class="show-more">…</a></span><span class="excerpt hidden">t need escaping when using exact matching of branch names containing these characters.

&lt;!--
Thank you for contributing to this project! You must fill out the information below before we can review this pull request. By explaining why you're making a change (or linking to a pull request) and what changes you've made, we can triage your pull request to the best possible team for review.

See our [CONTRIBUTING.md](/main/CONTRIBUTING.md) for information how to contribute.

For changes to content in [site policy](https://github.com/github/docs/tree/main/content/github/site-policy), see the [CONTRIBUTING guide in the site-policy repo](https://github.com/github/site-policy/blob/main/CONTRIBUTING.md).

We cannot accept changes to our translated content right now. See the [contributing.md](/main/CONTRIBUTING.md#earth_asia-translations) for more information.

Thanks again!
--&gt;

### Why:

Closes #8106

### What's being changed:

It's a very minimal change with just a few sentences, please see changed files.

### Check off the following:

- [ ] I have reviewed my changes in staging (look for the latest deployment event in your pull request's timeline, then click **View deployment**).
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/CONTRIBUTING.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")</span></p>
  </div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
